tests/pytests/integration/modules/test_pillar.py

```diff
@@ -478,3 +478,118 @@ def test_pillar_refresh_pillar_scheduler(salt_cli, salt_minion):
 
     # Refresh pillar once we're done
     salt_cli.run("saltutil.refresh_pillar", wait=True, minion_tgt=salt_minion.id)
+
+
+@pytest.mark.slow_test
+def test_pillar_match_filter_by_minion_id(salt_cli, salt_minion):
```

Comment:
These are actually 3 tests, which we can parametrize with pytest.

Reply:
I'm cool merging like this on the freeze branch, we should fix on the master branch though

Reply:
K I'll leave a not on the master PR

Reply:
Nevermind, its already merged. I'll prep a PR for master.